docs/repo/debugging/async-test-hangs.md

```diff
@@ -0,0 +1,270 @@
+```
+
+## Determining the executing test
+
+We need to find the xUnit types in memory that track test exception, in this case; [InvokeTestAsync](https://github.com/xunit/xunit/blob/9d10262a3694bb099ddd783d735aba2a7aac759d/src/xunit.execution/Sdk/Frameworks/Runners/XunitTestRunner.cs#L67). It is an async method, so we're going to find the blocked tests by looking through instances of its async state machine.
+
+```
+> !dumpheap -type InvokeTestAsync
+```
+
+```
+ Address       MT     Size
+036c8cb0 08389c3c       44     
+037a1828 08389c3c       44     
+039d9c40 08389c3c       44     
+03adb7c4 08389c3c       44     
+03b9cf68 08389c3c       44     
+03bd7638 08389c3c       44     
+03bf5748 08389c3c       44     
+03d75740 08389c3c       44     
+03d7fee0 08389c3c       44     
+03dc019c 08389c3c       44     
+03dc2e9c 08389c3c       44     
+03dd51a4 08389c3c       44     
+03ee8c00 08389c3c       44     
+03f98a38 08389c3c       44     
+03ff07e4 08389c3c       44     
+0bff4e9c 08389c3c       44     
+0c037b98 08389c3c       44     
+0c05c2b0 08389c3c       44     
+
+Statistics:
+      MT    Count    TotalSize Class Name
+08389c3c       18          792 Xunit.Sdk.XunitTestRunner+<InvokeTestAsync>d__4
+Total 18 objects
+```
+
+The output tells me there's 18 instances of the async state machine in memory. What we are looking for are state machines that have not completed.
+
+**Note:** *If you have Prefer DML (Command -> Prefer DML) turned on, the Address and MT (method table) column should be hyperlinked - you can just click the links instead of manually typing the commands.*
+
+Let's dump the first one:
+
+```
+> !DumpObj /d 036c8cb0
+```
+```
+Name:        Xunit.Sdk.XunitTestRunner+<InvokeTestAsync>d__4
+MethodTable: 08389c3c
+EEClass:     083b7c8c
+Size:        44(0x2c) bytes
+File:        E:\project-system2\artifacts\Debug\bin\UnitTests\xunit.execution.desktop.dll
+Fields:
+      MT    Field   Offset                 Type VT     Attr    Value Name
+7243f2dc  4000237       14         System.Int32  1 instance       -2 <>1__state
+015c84d8  4000238       18 ...lib]], mscorlib]]  1 instance 037a1840 <>t__builder
+08383de4  4000239        4 ...k.XunitTestRunner  0 instance 036b0ca0 <>4__this
+07869498  400023a        8 ...ceptionAggregator  0 instance 036b2d3c aggregator
+7243d488  400023b        c        System.String  0 instance 03501228 <output>5__2
+08389e5c  400023c       10 ....TestOutputHelper  0 instance 00000000 <testOutputHelper>5__3
+015c8dc4  400023d       24 ...cimal, mscorlib]]  1 instance 037a184c <>u__1
+
+```
+
+The value of the `<>1__state` field is what we're interested in, this represents the "current state" of the state machine:
+
+Value|Meaning
+---:|---
+-2|Finished executing
+-1|Not started or currently excecuting (should be active call stack)
+>=0| Blocked in an await. The number indicates the zero-based ordinal of which await in the method is currently waiting. 
```

Comment:
Markdown formatter isn't liking this line, its not part of the table and rendering as a quote. Maybe try &gt;?